src/Arcus.EventGrid/EventGridMessage.cs

```diff
@@ -1,11 +1,14 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using Arcus.EventGrid.Contracts.Interfaces;
 using GuardNet;
 
 namespace Arcus.EventGrid
 {
-    public class EventGridMessage<TEvent> where TEvent : IEvent, new()
+    /// <summary>
+    /// Representation of an message received from event grid containing a <see cref="SessionId"/> and a series of <see cref="Events"/>.
+    /// </summary>
+    /// <typeparam name="TEvent">The type of the event that was received.</typeparam>
+    public class EventGridMessage<TEvent>
```

Comment:
Ok, I'll just rename this? Will break some things maybe for users.

Reply:
We are targeting a new major version so breaking is ok! What name do you prefer?

Reply:
The `EentGridEventBatch` is a nice one I think.

Reply:
Sounds good to me!

Reply:
Just adding my 2 cents:  Isn't `EventGridEventBatch` too long ?  Since we're already in the `Arcus.EventGrid` namespace, it might be better to name this class `EventBatch` ?  (just a suggestion).

Reply:
I have the bad habbit of providing long names as we always use usings for namespace imports. When reading code you then don't have the context of where it comes from.

That said, I think `EventBatch` is also fine for me and clear, I don't have a preference really.

Reply:
Hmm, diffucult one.
The name `EventGridEventBatch` is not about the namespace `EventGrid` but about the event `EventGridEvent`.
So the name `EventGridEvent` + `Batch` alligns here I think, where as `EventBatch` is maybe to short. If all the other types here also would have a bit shorter name (like @tomkerkhove  said), it could be benitital, though.

Reply:
True, `EventGridEvent` should then be named `Event` for instance.  It's just a suggestion offcourse.

Reply:
`Event` & `EventBatch` is also fine for me, don't have a preference.

What do you think @stijnmoreels 	?

Reply:
In the context of other types whithin this repo, I don't think it would help beginners and other devs by using smaller names in this case.

Reply:
Let's stick with what we have then, it doesn't harm anybody and feels better for a library indeed.